From 0d83b349fa7340475406b2fe933c7467e4584091 Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@suse.de>
Date: Wed, 15 Mar 2023 11:44:24 +0100
Subject: [PATCH] getlogin_r: fix missing fallback if loginuid is unset (bug
 30235)

When /proc/self/loginuid is not set, we should still fall back to using
the traditional utmp lookup, instead of failing right away.
---
 sysdeps/unix/sysv/linux/getlogin_r.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/getlogin_r.c b/sysdeps/unix/sysv/linux/getlogin_r.c
index 879df85a16..4ae9a53503 100644
--- a/sysdeps/unix/sysv/linux/getlogin_r.c
+++ b/sysdeps/unix/sysv/linux/getlogin_r.c
@@ -59,10 +59,7 @@ __getlogin_r_loginuid (char *name, size_t namesize)
      value of, (uid_t) -1, so check if that value is set and return early to
      avoid making unneeded nss lookups. */
   if (uid == (uid_t) -1)
-    {
-      __set_errno (ENXIO);
-      return ENXIO;
-    }
+    return -1;
 
   struct passwd pwd;
   struct passwd *tpwd;
-- 
2.40.0

